snowfall/models/contextnet.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+
+# Copyright (c)  2021  University of Chinese Academy of Sciences (author: Han Zhu)
+# Apache 2.0
+
+
+import torch
+from torch import Tensor, nn
+from snowfall.models import AcousticModel
+from snowfall.models.conformer import Swish
+
+
+class ContextNet(AcousticModel):
+    """ContextNet. Reference: https://arxiv.org/pdf/2005.03191.pdf
+
+    Args:
+        num_features (int): Number of input features
+        num_classes (int): Number of output classes
+        kernel_size (int): Kernel size of convolution layers (default 3).
+        num_blocks (int): Number of context block (default 23).
+        num_layers (int): Number of depthwise convolution layers for each 
+                context block (except first and last block) (default 5).
+        alpha (float): The factor to scale the output channel of the network (default 1).
+        dropout (float): Dropout (default 0.1).
+    """
+
+    def __init__(
+        self,
+        num_features: int,
+        num_classes: int,
+        kernel_size: int = 3,
+        num_blocks: int = 23,
+        num_layers: int = 5,
+        alpha: int = 1,
+        dropout: int = 0.1,
+    ):
+        super().__init__()
+
+        self.num_features = num_features
+        self.num_classes = num_classes
+        self.subsampling_factor = 8
```

Comment:
I thought you said you were using 4, not 8?  I don't see you overriding this...
8 is too long for our speech units, which are phones.

Reply:
It's a typo and this variable is not used actually. The current result is using 4 subsampling.